### PR TITLE
Development

### LINKED ISSUE #8: Show a Reply button to TL1 on all topics that TL2+ can reply to, and redirect them to post approval
This is a longer-term issue to solve. Not high priority right now.

CTRL+F for "TODO: REPLIES:" in code.

We should consider putting the link in a different post than the first post, so that we don't need to change the code that handles creating a new topic/reply out of the first post in the message. We could put it into the system response and then parse it back from there, or we can attach the target topic as a custom_field on the message.

Make sure to also finish the prediction for target category on the front-end after implementing this.

### LINKED ISSUE #11: Try remove need of serializing is_post_approval to current_user
We could do the check on the client if we can figure out how to tunnel the data to the .hbs file

### PR DESCRIPTION
**Pull request by buildthomas**
Fixes #8 
Fixes #11

### CODE REVIEW

File: assets/javascripts/discourse/controllers/move-post-approval.js.es6 [hunk 1]
```diff
@@ -53,45 +57,62 @@ export default Controller.extend(ModalFunctionality, {
...
-        // TODO: use settings rather than hardcoding this?
         if (predictedTopicName.includes("Reply to Topic")
-         || predictedTopicName.includes("Request to Reply")
-         || predictedTopicName.includes(Discourse.SiteSettings.post_approval_redirect_reply_prefix))
+        || predictedTopicName.includes("Request to Reply")
+        || predictedTopicName.includes(Discourse.SiteSettings.post_approval_redirect_reply_prefix))
             predictedSelection = "existing_topic";
+        
+        const firstPost = this.get("model.postStream.posts")[0];
+        console.log(firstPost.get("pa_target_topic_id"));
```
Comment (Kampfkarren) [inline assets/javascripts/discourse/controllers/move-post-approval.js.es6]: debug line

File: assets/javascripts/discourse/controllers/move-post-approval.js.es6 [hunk 2]
```diff
@@ -53,45 +57,62 @@ export default Controller.extend(ModalFunctionality, {
...
+        predictedSelectedTopicId = parseInt(firstPost.get("pa_target_topic_id")); // grab from marker
 
-        const category = Category.list().forEach(
+        Category.list().forEach(
```
Comment (Kampfkarren) [inline assets/javascripts/discourse/controllers/move-post-approval.js.es6]: use `for (let c of Category.list()) {` instead

File: plugin.rb [hunk 3]
```diff
@@ -64,12 +67,13 @@ def can_move_topic_to_category?(category)
...
   module PostAlerterInterceptor
     def is_redirectable_topic(topic)
+      category = topic.category
       return SiteSetting.post_approval_enabled &&
         SiteSetting.post_approval_redirect_enabled &&
         SiteSetting.post_approval_redirect_group.present? &&
         !(topic.custom_fields["post_approval"]) && # suppress notifications unless post was already approved
         topic.user&.trust_level <= SiteSetting.post_approval_redirect_tl_max &&
-        topic.category&.pa_redirect_topic_enabled
+        category && category.pa_redirect_topic_enabled
```
Comment (Kampfkarren) [inline plugin.rb]:
```suggestion
        category&.pa_redirect_topic_enabled
```

File: assets/javascripts/discourse/templates/connectors/connectors/topic-footer-main-buttons-before-create/post-approval-button.js.es6 [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export default {
+  setupComponent(args, component) {
+    // Find a group with setting name in the user's groups:
+    const group = Discourse.User.currentProp("groups").find(
+        g => g.name.toUpperCase() === Discourse.SiteSettings.post_approval_button_group.toUpperCase()
+    );
+    // Set visibility of button based on whether group was found:
+    component.set("isPostApproval", group != undefined);
```
Comment (Kampfkarren) [inline assets/javascripts/discourse/templates/connectors/connectors/topic-footer-main-buttons-before-create/post-approval-button.js.es6]:
```suggestion
    component.set("isPostApproval", group !== undefined);
```

File: plugin.rb (hunk 3 above)
Comment (buildthomas) [reply plugin.rb]: Ignoring this one since it doesn't work for w/e reason

**Pull request merged**